### DISCUSSION TITLE
请问怎么样在RunTimeLayoutConfig的menu加载远程数据时，动态创建路由而不是在routes.ts里固定配置？

### DISCUSSION THREAD

**Discussion by whiteshader** [umijs/umi#7911] [General]
请问怎么样在RunTimeLayoutConfig的menu加载远程数据时，动态创建路由？在routes.ts里里固定配置这样后台生成的菜单做不到动态URL。

**Comment by perfect** [comment]
我也有这个问题，菜单选项是数据库返回的，怎么动态实现

**Reply by xXAvoraXx** [reply]
https://github.com/umijs/umi/discussions/11770

**Reply by perfect** [reply]
这个是一直有的功能，还是刚刚开发的

**Comment by fz6m** [comment]
如果要动态路由表，可以通过运行时 [`patchClientRoutes`](https://umijs.org/docs/api/runtime-config#patchclientroutes-routes-) 修改路由来实现，一个例子见：https://github.com/umijs/umi/discussions/9717#discussioncomment-4078053

**Comment by xXAvoraXx** [comment]
由于 umi 的静态路由定义无法解决我们的问题，因此我们需要 api 的动态路由定义。我已经开始为这种情况开发一个插件，我们可以一起开发。

https://github.com/umijs/umi/discussions/11770

**Comment by whiteshader** [comment]
本问题已经解决，具体代码在以下项目。
https://github.com/whiteshader/React-Admin-Cloud
